### ISSUE TITLE
Wrongly played MSX2 track

### ISSUE THREAD

**Issue by vitamin-caig** [ValleyBell/libvgm#106]
Here the full pack: https://vgmrips.net/packs/pack/snatcher-msx2

[Twilight of Neo Kobe City](https://vgmrips.net/packs/pack/snatcher-msx2#05-twilight-of-neo-kobe-city) track sounds different via libvgm and site player. 

Here the reference: https://www.youtube.com/watch?v=KbbF9odLDxI

**Comment by ValleyBell** [comment]
Hmm...this is actually a bug in the VGM.  
The VGM sets the "SCC+" (K052539) mode flag, but it assumes that channels 4+5 share their wave RAM. (SCC non-plus/K051649 mode)

This used to work fine in the old VGMPlay due to an implementation oddity.
The actual solution should be to fix the VGM.

**Comment by ValleyBell** [comment]
The rip itself should be mostly okay. There is just a wrong flag set in the VGM header, but this can be easily patched afterwards.

I'll probably go further and just check+update all the SCC packs though.

**Comment by grauw** [comment]
Snatcher is an SCC+ game, so the flag is correctly set, but it sounds like the commands used are wrong.

**Comment by ValleyBell** [comment]
Okay, so then the pack is bad. I checked a few tracks and they all write only to the wave RAM of the first 4 channels. (using the SCC non-plus commands)

---

This issue has one good side though: This confirms that libvgm now behaves like VGMPlay MSX with incorrect SCC/SCC+ flags. (which is a good thing, as it makes incorrect VGM rips more obvious)

**Comment by grauw** [comment]
Ping @sdsnatcher73.

**Comment by sdsnatcher73** [comment]
Most likely the issue is then originated in NEZplug v0.9.4.8 + 2 + 19.20 which was used to create the VGM's

**Comment by sdsnatcher73** [comment]
[01.vgz.zip](https://github.com/ValleyBell/libvgm/files/10527296/01.vgz.zip)
Here is the unmodified vgz as ripped with NEZplug, for analysis.

**Comment by sdsnatcher73** [comment]
I guess kss2vgm (used for SD Snatcher) does log correctly, but officially we're not supposed to use that, right?

**Comment by sdsnatcher73** [comment]
New version logged with kss2vgm. Please let me know if this works better...
[Snatcher (MSX2, PSG+SCC-I).zip](https://github.com/ValleyBell/libvgm/files/10543530/Snatcher.MSX2.PSG%2BSCC-I.zip)

**Comment by vitamin-caig** [comment]
> New version logged with kss2vgm. Please let me know if this works better... [Snatcher (MSX2, PSG+SCC-I).zip](https://github.com/ValleyBell/libvgm/files/10543530/Snatcher.MSX2.PSG%2BSCC-I.zip)

Yes, looks like problem is solved. At least, I can't hear any significant difference between reference and updated version

**Comment by ValleyBell** [comment]
The pack on vgmrips is fixed now. (see [update post](https://vgmrips.net/forum/viewtopic.php?p=19289#p19289))
